CLIENT_CONFIGURATION.md

```diff
@@ -0,0 +1,155 @@
+
+The configuration method depends on whether you are using the `grpc` (default) or `rest` transport.
+
+### Proxy with gRPC
+
+When using the gRPC transport, the client library respects the [standard environment variables](https://grpc.github.io/grpc/php/md_doc_environment_variables.html). You **do not** need to configure this in the PHP code itself.
+
+Set the following environment variables in your shell or Docker container:
+
+```
+export http_proxy="http://proxy.example.com:3128"
+export https_proxy="http://proxy.example.com:3128"
+```
+
+**Handling Self-Signed Certificates (gRPC):** If your proxy uses a self-signed certificate (Deep Packet Inspection), you cannot simply "ignore" verification in gRPC. You must provide the path to the proxy's CA certificate bundle.
+
+```
+# Point gRPC to a CA bundle that includes your proxy's certificate
+export GRPC_DEFAULT_SSL_ROOTS_FILE_PATH="/path/to/roots.pem"
+```
+
+### Proxy with REST
+
+If you are forcing the `rest` transport (or using a library that only supports REST), you must configure the proxy via the `transportConfig` option. This passes the settings down to the underlying Guzzle client.
+
+```php
+use Google\Auth\HttpHandler\HttpHandlerFactory;
+use Google\Cloud\SecretManager\V1\Client\SecretManagerServiceClient;
+
+$httpClient = new GuzzleHttp\Client([
+    // Standard Guzzle proxy configuration
+    'proxy' => 'http://user:password@proxy.example.com',
+    // (Optional) Disable SSL Verification (Development Only)
+    // 'verify' => false
+]);
+$httpHandler = HttpHandlerFactory::build($httpClient);
+$secretManagerClient = new SecretManagerServiceClient([
+    'transport' => 'rest',
+    'transportConfig' => ['rest' => ['httpHandler' => [$httpHandler, 'async']]],
+]);
+```
+
+## 4\. Configuring Retries and Timeouts
+
+There are two ways to configure retries and timeouts: global client configuration (complex) and per-call configuration (simple).
+
+### Per-Call Configuration (Recommended)
+
+For most use cases, it is cleaner to override settings for specific calls using `Google\ApiCore\Options\CallOptions` (or the `$optionalArgs` array in generated clients).
+
+#### Available `retrySettings` Keys
+
+When passing an array to `retrySettings`, you can use the following keys to fine-tune the exponential backoff strategy:
+
+| Key | Type | Description |
+| ----- | ----- | ----- |
+| `retriesEnabled` | `bool` | Enables or disables retries for this call. |
+| `maxRetries` | `int` | The maximum number of retry attempts. |
+| `initialRetryDelayMillis` | `int` | Wait time before the *first* retry (in ms). |
+| `retryDelayMultiplier` | `float` | Multiplier applied to the delay after each failure (e.g., `1.5`). |
+| `maxRetryDelayMillis` | `int` | The maximum wait time between any two retries. |
+| `totalTimeoutMillis` | `int` | Total time allowed for the request (including all retries) before giving up. |
+
+#### Example: Advanced Backoff
+
+```php
+// Advanced Retry Configuration
```

Comment:
Having these docs would've save me a lot of time for my unit test on the RequestIdMiddleware haha.